tests/src/Traits/OgTrait.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+namespace Drupal\joinup\Traits;
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\og\Entity\OgMembership;
+use Drupal\og\Og;
+use Drupal\og\OgMembershipInterface;
+
+/**
+ * Contains helper methods regarding the organic groups.
+ *
+ * @package src\Traits
+ */
+trait OgTrait {
+
+  /**
+   * User with a given og role is already logged in.
+   *
+   * If the user has more than the required roles, he might have permissions
+   * from the rest of the roles that will lead the test to a false positive.
+   * For this reason, we request check for the specific roles passed.
+   *
+   * @param array $roles
+   *   An array of roles to check.
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   A single role, or multiple comma-separated roles in a single string.
+   *
+   * @return bool
+   *   Returns TRUE if the current logged in user has this role (or roles).
+   */
+  protected function loggedInWithOgRole($roles, EntityInterface $entity) {
```

Comment:
Who writes these awful documentation? The $entity object is the group. Updated the doc and renamed the $entity to $group... This documentation was crime against humanity.

For the record, initially, I started to implement it without this method. I did not care that much if I had to force the recreation of a user every time. But I saw that the drupal context actually checks if the current user has the given role before creating a new one so I thought I should mimic it.